aep/general/0158/aep.md.j2

```diff
@@ -1,100 +1,103 @@
+      child_type: "library.googleapis.com/Book"
+    }];
 
   // The maximum number of books to return. The service may return fewer than
   // this value.
   // If unspecified, at most 50 books will be returned.
   // The maximum value is 1000; values above 1000 will be coerced to 1000.
-  maxPageSize: bigint;
+  int32 page_size = 2;
 
   // A page token, received from a previous `ListBooks` call.
   // Provide this to retrieve the subsequent page.
   //
   // When paginating, all other parameters provided to `ListBooks` must match
   // the call that provided the page token.
-  pageToken: string;
+  string page_token = 3;
 }
 
 // The response structure from listing books.
-interface ListBooksResponse {
+message ListBooksResponse {
   // The books from the specified publisher.
-  books: Book[];
+  repeated Book books = 1;
 
   // A token that can be sent as `page_token` to retrieve the next page.
   // If this field is omitted, there are no subsequent pages.
-  nextPageToken: string;
+  string next_page_token = 2;
 }
 ```
 
-- Request definitions for collections **should** define an
-  `int32 max_page_size` field, allowing users to specify the maximum number of
-  results to return.
-  - If the user does not specify `max_page_size` (or specifies `0`), the API
+{% tab oas %}
+
+**Note:** OAS example not yet written.
+
+{% endtabs %}
+
+- Request messages for collections **should** define an integer (`int32` for
+protobuf) `page_size` field, allowing users to specify the maximum number of
+results to return.
+  - If the user does not specify `page_size` (or specifies `0`), the API
     chooses an appropriate default, which the API **should** document. The API
     **must not** return an error.
-  - If the user specifies `max_page_size` greater than the maximum permitted by
-    the service, the service **should** coerce down to the maximum permitted
-    page size.
-  - If the user specifies a negative value for `max_page_size`, the API
-    **must** return a `400 Bad Request` error.
-  - The service **should** the number of results requested, unless the end of
-    the collection is reached.
-    - However, occasionally this is infeasible, especially within expected time
-      limits. In these cases, the service **may** return fewer results than the
-      number requested (including zero results), even if not at the end of the
-      collection.
-- Request definitions for collections **should** define a `string page_token`
+  - If the user specifies `page_size` greater than the maximum permitted by the
+    API, the API **should** coerce down to the maximum permitted page size.
+  - If the user specifies a negative value for `page_size`, the API **must**
+    send an `INVALID_ARGUMENT` error.
+  - The API **may** return fewer results than the number requested (including
+    zero results), even if not at the end of the collection.
+- Request messages for collections **should** define a `string page_token`
```

Comment:
```suggestion
- Request schemas for collections **should** define a `string page_token`
```

"messages" is proto-specific.  "definitions" works but I think we've used "schemas" elsewhere.